tmlibrary/tmlib/tools/base.py

```diff
@@ -465,8 +465,14 @@ def register_result(self, submission_id, mapobject_type_name,
                 filter_by(submission_id=submission_id).\
                 one_or_none()
             if result is None:
+                # Define the name. If a name is set in the input args, that's the 
+                # name. Otherwise, it's the name of the submitting class
+                if 'name' in result_attributes.keys():
+                    name = result_attributes['name']
+                else:
+                    name = self.__class__.__name__
```

Comment:
These could become:
```suggestion
                name = result_attributes.get('name', self.__class__.__name__)
```
if `result_attributes` is a "classic" dictionary, see: https://docs.python.org/3/library/stdtypes.html#dict.get